### PR TITLE
genpolicy: support readonly hostpath

### PR DESCRIPTION
**Pull request by Redent0r**
Support readonly hostpath and readonly  emptyDir mount

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
lgtm, gracias @Redent0r!

File: tests/integration/kubernetes/runtimeclass_workloads/k8s-policy-rc.yaml [hunk 1]
```diff
@@ -27,6 +27,12 @@ spec:
         volumeMounts:
           - name: host-empty-vol
             mountPath: "/host/cache"
+          - mountPath: /tmp/results
+            name: hostpath-vol
+            readOnly: true
```
Comment (sprt) [inline tests/integration/kubernetes/runtimeclass_workloads/k8s-policy-rc.yaml:33]: Should we also add a read-write host mount to ensure that path is tested as well?

File: tests/integration/kubernetes/runtimeclass_workloads/k8s-policy-rc.yaml [hunk 2]
```diff
@@ -27,6 +27,7 @@ spec:
         volumeMounts:
           - name: host-empty-vol
             mountPath: "/host/cache"
+            readOnly: true
```
Comment (sprt) [inline tests/integration/kubernetes/runtimeclass_workloads/k8s-policy-rc.yaml:30]: Should we rather add a new read-only emptyDir mount to ensure both paths are tested?

**Review by sprt — APPROVED** [review]
LGTM - left some comments about coverage.

File: tests/integration/kubernetes/runtimeclass_workloads/k8s-policy-rc.yaml (hunk 1 above)
Comment (Redent0r) [reply tests/integration/kubernetes/runtimeclass_workloads/k8s-policy-rc.yaml:33]: Good call. I'll add readOnly and non readOnly for both.

Comment (Redent0r) [reply tests/integration/kubernetes/runtimeclass_workloads/k8s-policy-rc.yaml]: Updated to test both paths for emptyDir and hostPath

**Review by GabyCT — APPROVED** [review]

**Pull request merged**